File: project_euler/problem_218/solution.py
"""
Problem 218: Perfect Right-angled Triangles

Every perfect right-angled triangle has area divisible by 84.
Therefore no perfect triangle can fail both the divisibility-by-6
and divisibility-by-28 tests.
"""


def count_non_super_perfect(limit):
    # The proof is independent of the hypotenuse bound.
    return 0


def solve():
    print(count_non_super_perfect(10**16))


if __name__ == "__main__":
    solve()

File: project_euler/problem_218/solution.cpp
#include <bits/stdc++.h>
using namespace std;

long long countNonSuperPerfect(long long /*limit*/) {
    // The proof shows every perfect right-angled triangle has area divisible by 84,
    // so none can fail both the divisibility-by-6 and divisibility-by-28 tests.
    return 0;
}

int main() {
    cout << countNonSuperPerfect(10000000000000000LL) << '\n';
    return 0;
}
